src/Common/Core/Impl/Collections/EnumerableExtensions.cs

```diff
@@ -113,5 +113,11 @@ public static class EnumerableExtensions {
                 }
             }
         }
+
+        public static void ForEach<T>(this IEnumerable<T> enumerable, Action<T> action) {
```

Comment:
`AsList` allocates more memory. All the argument is really about readability and as Eric said 'philosophical objections'